server/src/main.cpp

```diff
@@ -94,6 +91,19 @@ int main(int argc, char **argv)
         return EXIT_FAILURE;
     }
 
+    int status = std::system(("sudo hostnamectl set-hostname " + newHostname).c_str());
```

Comment:
Write comments about what is going on here

Reply:
Done